deploy-argocd-quicklab.sh

```diff
@@ -0,0 +1,56 @@
+API=upi-0.mptest.lab.upshift.rdu2.redhat.com:6443
+KUBEPASS=rSszg-hYQnn-h9zsi-aAoEb
+
+EMP="\e[1;4m"
+NORMAL="\e[0m"
+
+if which kustomize >/dev/null; then
+  kustomize=kustomize
+else
+  kustomize="toolbox run --container of-toolbox-v0.1.0 kustomize"
+fi
+
+if ! $kustomize --help > /dev/null; then
+  echo -e "${EMP}kustomize is not available --> aborting${NORMAL}"
+  exit 1
+fi
+
+echo -e "${EMP}Checking for GPG key${NORMAL}"
+gpg --list-keys john@doe.com || base64 -d < examples/key.asc | gpg --import
+
+oc login -u kubeadmin -p $KUBEPASS $API
+
+echo -e "${EMP}Creating admin user 'myadmin'${NORMAL}"
+htpasswd -nb myadmin foobar68 > /tmp/oc.htpasswd
+oc create secret generic htpass-secret --from-file=htpasswd=/tmp/oc.htpasswd -n openshift-config
+cat <<EOF | oc apply -f -
+apiVersion: config.openshift.io/v1
+kind: OAuth
+metadata:
+  name: cluster
+spec:
+  identityProviders:
+  - name: my_htpasswd_provider
+    mappingMethod: claim
+    type: HTPasswd
+    htpasswd:
+      fileData:
+        name: htpass-secret
+EOF
+
+oc adm policy add-cluster-role-to-user cluster-admin myadmin
```

Comment:
unfortunately this this can't work by itself. Openshift creates the `user` entity only after it logged in for the first time for all OAuth providers. You need to "login" as the target user before this command.

Reply:
hmm, interesting. you are surely right, but I have run this several time successfully. Might be I run have run it multiple times and there's the login command a few lines lower. So I just skip the 2.

Reply:
Swapped.